flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/FailoverStrategy.java

```diff
@@ -34,22 +34,4 @@
 	 * @return set of IDs of vertices to restart
 	 */
 	Set<ExecutionVertexID> getTasksNeedingRestart(ExecutionVertexID executionVertexId, Throwable cause);
-
-	// ------------------------------------------------------------------------
-	//  factory
-	// ------------------------------------------------------------------------
-
-	/**
-	 * The factory to instantiate {@link FailoverStrategy}.
-	 */
-	interface Factory {
```

Comment:
Don't we need this interface again in the future?

Reply:
So far I think there is no need to create a JM shared FailoverStrategy factory. Each JM/Scheduler can create their own failoverStrategy directly.
We should implement a loader util for creating failoverStrategy from ExecutionGraph though.
WDYT?

Reply:
I cannot give you a definite answer right now but it seems to be that deleting the `Factory` interface is not needed for the adapter.  I think we can make a decision once the default `SchedulerNG` implementation is done.

Reply:
Ok. Let's exclude this change for now.